device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/twin/TwinMetadata.java

```diff
@@ -0,0 +1,269 @@
+ *          },
+ *          "NewValue":{
+ *              "$lastUpdated":"2017-09-21T02:07:44.238Z",
+ *              "$lastUpdatedVersion":5
+ *          }
+ *      }
+ *  }
+ * }
+ * </pre>
+ *
+ * @see <a href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-device-twins">Understand and use device twins in IoT Hub</a>
+ * @see <a href="https://docs.microsoft.com/en-us/rest/api/iothub/devicetwinapi">Device Twin Api</a>
+ */
+// Unchecked casts of Maps to Map<String, Object> are safe as long as service is returning valid twin json payloads. Since all json keys are Strings, all maps must be Map<String, Object>
+@SuppressWarnings("unchecked")
+public class TwinMetadata
+{
+    // the entity last updated date and time in the TwinCollection
+    public static final String LAST_UPDATE_TAG = "$lastUpdated";
+    private Date lastUpdated;
+
+    // the entity last updated version in the TwinCollection
+    public static final String LAST_UPDATE_VERSION_TAG = "$lastUpdatedVersion";
+    private final Integer lastUpdatedVersion;
+
+    // the entity last updated by in the TwinCollection. Value is a configuration applied on the Twin.
+    public static final String LAST_UPDATED_BY = "$lastUpdatedBy";
+    private String lastUpdatedBy;
+
+    // the entity last updated by digest in the TwinCollection which represents service internal version for the applied configuration.
+    public static final String LAST_UPDATED_BY_DIGEST = "$lastUpdatedByDigest";
+    private String lastUpdatedByDigest;
+
+    /**
+     * CONSTRUCTOR
+     *
+     * <p> This private constructor will receive and store the metadata parameters.
+     *
+     * @param lastUpdated        the {@code String} with the date and time UTC of the last update on the entity. It can be {@code null}, empty or invalid.
+     * @param lastUpdatedVersion the {@code Integer} with the version of the last update on the entity. It can be {@code null}.
+     * @throws IllegalArgumentException If no valid parameter was provide and the class will be empty, or if the DateTime is invalid.
+     */
+    TwinMetadata(String lastUpdated, Integer lastUpdatedVersion, String lastUpdatedBy, String lastUpdatedByDigest) {
+        if (!Tools.isNullOrEmpty(lastUpdated)) {
+            /* SRS_TWIN_METADATA_21_001: [The constructor shall parse the provided `lastUpdated` String to the Date and store it as the TwinMetadata lastUpdated.] */
+            /* SRS_TWIN_METADATA_21_002: [The constructor shall throw IllegalArgumentException if it cannot convert the provided `lastUpdated` String to Date.] */
+            this.lastUpdated = ParserUtility.getDateTimeUtc(lastUpdated);
+        }
+
+        if (lastUpdatedBy != null) {
+            this.lastUpdatedBy = lastUpdatedBy;
+        }
+
+        if (lastUpdatedByDigest != null) {
+            this.lastUpdatedByDigest = lastUpdatedByDigest;
+        }
+
+        /* SRS_TWIN_METADATA_21_003: [The constructor shall store the provided lastUpdatedVersion as is.] */
```

Comment:
Do you want to remove these legacy comments?

Reply:
Gladly